torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1728,19 +1730,58 @@ struct to_ir {
   // raise a
   //
   // We ignore the expression following raise
-  void emitRaise(const SourceRange& loc) {
-    const std::string exception = "Exception";
-    auto string_input = insertConstant(*graph, exception, loc);
-    graph->insert(prim::RaiseException, {string_input}, {}, loc);
+  void emitRaise(const Raise& raise) {
+    auto sv = emitSugaredExpr(raise.expr(), 1);
+    Value* value = nullptr;
+
+    auto exception_instance = std::dynamic_pointer_cast<ExceptionMessageValue>(sv);
+    if (auto exception_instance =
+            std::dynamic_pointer_cast<ExceptionMessageValue>(sv)) {
+      // The typical case, an instance of the exception class was thrown:
+    //    raise RuntimeError("error")
+      value = exception_instance->getValue();
```

Comment:
Can we: have `ExceptionMessageValue` store all of the values it is called with as a list of values. Then, here, try to cast each one to str ? ideally with `str` cast value above.

Reply:
This is now handled by making the sugared value construct a tuple